src/lib.rs

```diff
@@ -188,12 +185,13 @@ impl Plugin for Triforce {
     type InitFeatures = ();
     type AudioFeatures = ();
 
-    fn new(_info: &PluginInfo, _features: &mut ()) -> Option<Self> {
+    fn new(info: &PluginInfo, _features: &mut ()) -> Option<Self> {
         Some(Self {
             hangle_curr: 0f32,
             vangle_curr: 0f32,
             freq_curr: 1000f32,
-            last_update: SystemTime::now(),
+            samples_since_last_update: u32::max_value(),
+            sample_rate: info.sample_rate() as f32,
```

Comment:
The graph sample rate can change. I'm pretty sure PipeWire tears down and reinits plugins when the graph rate changes but it would be good to make this robust against the case that it doesn't.

Reply:
I am fairly sure that the sample rate is fixed for the lifetime of an LV2 plugin. So PipeWire will have to teardown at least the LV2 plugins, other processors might be more expressive.
There is nothing we can do about that.